src/net_processing.cpp

```diff
@@ -657,25 +663,56 @@ static void FindNextBlocksToDownload(NodeId nodeid, unsigned int count, std::vec
     }
 }
 
-void RequestTx(CNodeState *state, uint256 &txid) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+void EraseTxRequest(uint256& txid) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+{
+    m_already_asked_for.erase(txid);
+}
+
+int64_t GetTxRequestTime(const uint256& txid)
+{
+    auto it = m_already_asked_for.find(txid);
+    if (it != m_already_asked_for.end()) {
+        return it->second;
+    }
+    return 0;
+}
+
+void UpdateTxRequestTime(const uint256 &txid, int64_t request_time)
 {
-    CNodeState::TxDownloadState &peer = state->m_tx_download;
-    if (peer.m_tx_announced.size() >= MAX_PEER_TX_ANNOUNCEMENTS || peer.m_tx_announced.count(txid)) {
+    auto it = m_already_asked_for.find(txid);
+    if (it == m_already_asked_for.end()) {
+        m_already_asked_for.insert(std::make_pair(txid, request_time));
+    } else {
+        m_already_asked_for.update(it, request_time);
+    }
+}
+
+
+void RequestTx(CNodeState *state, const uint256 &txid) {
+    CNodeState::TxDownloadState &peer_download_state = state->m_tx_download;
+    if (peer_download_state.m_tx_announced.size() >= MAX_PEER_TX_ANNOUNCEMENTS || peer_download_state.m_tx_announced.count(txid)) {
         // Too many queued announcements from this peer, or we already have
         // this announcement
         return;
     }
-    peer.m_tx_announced.insert(txid);
 
-    int64_t process_time = GetTimeMicros();
+    int64_t process_time;
+    int64_t last_request_time = GetTxRequestTime(txid);
+    // First time requesting this tx
+    if (last_request_time == 0) {
+        peer_download_state.m_tx_announced.insert(txid);
+        process_time = GetTimeMicros();
+    } else {
+        // Add extra delay so that first peer from the list is not prioritizied due to ThreadMessageHandler delay
```

Comment:
nit: prioritized

Also I find this comment a bit confusing, perhaps just something like:
`// Randomize the delay to avoid biasing some peers over others (such as due to fixed ordering of peer processing in ThreadMessageHandler)`

(BTW sorry for writing this code incorrectly the first time, I mistakenly thought we shuffled our peers each loop iteration in ThreadMessageHandler, which is why I didn't add randomization here in the first place!)